Teleport was constantly asking for username+HOTP+password

kontsevoy opened the pull request:
The problem was in `time.Unix()` instead of `time.Unixnano()`
I'm not even sure how Unix() got in there...

kontsevoy committed:
Teleport was constantly asking for username+HOTP+password

The problem was in `time.Unix()` instead of `time.Unixnano()`
I'm not even sure how Unix() got in there...

kontsevoy commented on lib/client/api.go:
```diff
@@ -94,9 +94,6 @@ type Config struct {
 
-	// AuthMethods if passed will be used to login into cluster
-	AuthMethods []ssh.AuthMethod
-
```
This field was never used.

klizhentas replied:
Can you bring it back? I use it in the library mode, when passing external authenticators (not local auth)

klizhentas commented:
lgtm except that I need this `AuthMethods` thing to exist

kontsevoy commented:
Mm... this can be greatly simplified. Why do you need a separate boolean flag for "skipLocalAuth" then? If the purpose of this to supply your own auth methods, why the need for two parameters?

kontsevoy committed:
Brought back client:Config.AuthMethods

But...

I removed another array of []ssh.AuthMethod from Teleport client itself
to avoid having a confusingly duplicated array.

kontsevoy commented:
Ok, I brought back the original Config.AuthMethods, but removed tc.authMethods to avoid duplication.

kontsevoy commented:
retest this

kontsevoy commented:
retest this please

klizhentas commented:
SkipLocalAuth tells client not to resort to local auth method if it failed everything else here:

https://github.com/gravitational/teleport/blob/master/lib/client/api.go#L704

Lgtm
